src/components/molecules/EventDisplay/EventDisplay.tsx

```diff
@@ -35,49 +31,58 @@ export const EventDisplay: React.FC<EventDisplayProps> = ({ event, venue }) => {
     "schedule-event-container--live": isLiveEvent,
   });
 
+  const starEventHour = formatHour(event.start_utc_seconds);
+
+  const oneHourBehind = format(
+    new Date().setHours(new Date().getHours() - 1),
+    "HH"
+  );
+  const duration = Math.floor(event.duration_minutes / 60);
+  const beginnigToShow = Number(starEventHour) - Number(oneHourBehind);
```

Comment:
Why do we format the numbers and then convert them back to numbers? Wouldn't it be better to use milliseconds to do math and then, when you got the result, format it to a string & use it

Reply:
Namings are not very consistent here either. 
1) eventStartInHours
2) oneHourBeforeEventStart
3) eventDuration

Need to get into the logic to get better namings

@0xdevalias maybe you could help here?

Reply:
There are also heaps of helpers in `src/utils/time.ts` and even more/probably better/more useful helpers in the `date-fns` lib, eg.

- https://date-fns.org/v2.20.1/docs/getHours
- https://date-fns.org/v2.20.1/docs/subHours
- etc

---

From a quick skim through the logic above:

- `oneHourBehind` is "one hour before the event starts"
- `duration` is "duration of the event in hours"
- `beginnigToShow` is (most likely, need to check where/how it's used) "start showing this from 1 hour before it starts"

Reply:
see the comment: https://github.com/sparkletown/sparkle/pull/1302#discussion_r628240676